src/main/java/com/pinterest/secor/reader/LegacyKafkaMessageIterator.java

```diff
@@ -99,7 +102,11 @@ public void init(SecorConfig config) throws UnknownHostException {
 
     @Override
     public void commit(TopicPartition topicPartition, long offset) {
-        mConsumerConnector.commitOffsets();
+        TopicAndPartition kafkaTopicPartition = new TopicAndPartition(topicPartition.getTopic(), topicPartition.getPartition());
+        kafka.common.OffsetAndMetadata offsetAndMetadata = OffsetAndMetadata.apply(offset);
+
+        LOG.info("committing {} offset {} to kafka", topicPartition, offset);
+        mConsumerConnector.commitOffsets(ImmutableMap.of(kafkaTopicPartition, offsetAndMetadata), true);
```

Comment:
What does the argument 'true' mean?

Reply:
It means that if the offsets commit will fail, there will be no retries after failing to commit offsets, which is consistent with the previous behaviour, since `mConsumerConnector.commitOffsets()` did the same thing.